api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/contexts"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability"
+	observ_utils "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability/utils"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/logger"
+)
+
+const loggerName = "providers.target.piccolo"
+
+var sLog = logger.NewLogger(loggerName)
+
+type PiccoloTargetProviderConfig struct {
+	Name string `json:"name"`
+	Url  string `json:"url"`
+}
+
+type PiccoloTargetProvider struct {
+	Config  PiccoloTargetProviderConfig
+	Context *contexts.ManagerContext
+}
+
+func PiccoloTargetProviderConfigFromMap(properties map[string]string) (PiccoloTargetProviderConfig, error) {
+	ret := PiccoloTargetProviderConfig{}
+	if v, ok := properties["name"]; ok {
+		ret.Name = v
+	}
+	return ret, nil
+}
+func (d *PiccoloTargetProvider) InitWithMap(properties map[string]string) error {
+	config, err := PiccoloTargetProviderConfigFromMap(properties)
+	if err != nil {
+		return err
+	}
+	return d.Init(config)
+}
+func (s *PiccoloTargetProvider) SetContext(ctx *contexts.ManagerContext) {
+	s.Context = ctx
+}
+
+func (d *PiccoloTargetProvider) Init(config providers.IProviderConfig) error {
+	_, span := observability.StartSpan("Piccolo Target Provider", context.TODO(), &map[string]string{
+		"method": "Init",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	sLog.Info("  P (Piccolo Target): Init()")
+
+	// convert config to PiccoloTargetProviderConfig type
+	piccoloConfig, err := toPiccoloTargetProviderConfig(config)
+	if err != nil {
+		sLog.Errorf("  P (Piccolo Target): expected PiccoloTargetProviderConfig: %+v", err)
+		return err
+	}
+
+	d.Config = piccoloConfig
+	return nil
+}
+
+func toPiccoloTargetProviderConfig(config providers.IProviderConfig) (PiccoloTargetProviderConfig, error) {
+	ret := PiccoloTargetProviderConfig{}
+	data, err := json.Marshal(config)
+	if err != nil {
+		return ret, err
+	}
+	err = json.Unmarshal(data, &ret)
+	return ret, err
+}
+
+func (i *PiccoloTargetProvider) Get(ctx context.Context, deployment model.DeploymentSpec, references []model.ComponentStep) ([]model.ComponentSpec, error) {
+	ctx, span := observability.StartSpan("Piccolo Target Provider", ctx, &map[string]string{
+		"method": "Get",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	sLog.InfofCtx(ctx, "  P (Piccolo Target): getting artifacts: %s - %s, traceId: %s", deployment.Instance.Spec.Scope, deployment.Instance.ObjectMeta.Name, span.SpanContext().TraceID().String())
+
+	ret := make([]model.ComponentSpec, 0)
+	for _, component := range references {
+		properties := component.Component.Properties
+		name := properties["workload.name"].(string)
+		workloadType := properties["workload.type"].(string)
+		req, err := http.NewRequest("GET", i.Config.Url+workloadType+"/"+name, nil)
```

Comment:
The Get() method and Apply() method should be consistent. For example, the Get() method gets by the path "workload.type/workload.name", while the Apply() method posts to path component.Name, which isn't necessary the same with workload.name. If workload.name and component.name are the same, probably we should exclude workload.name in properties and use component.Name in Get()

Reply:
Also, the Apply() method doesn't use workload.type. Maybe workload.type should be included in the request properties?